.github/workflows/kserve_m2m_test.yaml

```diff
@@ -162,58 +155,240 @@ jobs:
+          -H "Authorization: Bearer ${KSERVE_M2M_TOKEN}" \
+          -H "Content-Type: application/json" \
+          "http://${KSERVE_INGRESS_HOST_PORT}/v1/models/isvc-sklearn:predict" \
+          -d '{"instances": [[6.8, 2.8, 4.8, 1.4]]}')
+
+        HTTP_CODE=${RESPONSE: -3}
+        RESPONSE_BODY=${RESPONSE:0:${#RESPONSE}-3}
+
+        if [[ "$HTTP_CODE" == "200" && "$RESPONSE_BODY" == *"predictions"* ]]; then
+          echo "Test passed: Valid token access successful with code $HTTP_CODE"
+        else
+          echo "Test failed: Valid token access returned $HTTP_CODE"
+          echo "Response: $RESPONSE_BODY"
+          exit 1
+        fi
+
+    # Test 2: External access without token, Must Fail, Redirects to login page
+    - name: Test external access without token
       run: |
         export KSERVE_INGRESS_HOST_PORT=localhost:8080
-        export KSERVE_M2M_TOKEN="$(kubectl -n kubeflow-user-example-com create token default-editor)"
 
-        echo "=== AuthorizationPolicy Details ==="
-        kubectl get authorizationpolicy -n kubeflow-user-example-com -o yaml
+        RESPONSE=$(curl -s -o /dev/null -w "%{http_code}" \
+          -H "Host: isvc-sklearn.kubeflow-user-example-com.example.com" \
+          -H "Content-Type: application/json" \
+          "http://${KSERVE_INGRESS_HOST_PORT}/v1/models/isvc-sklearn:predict" \
+          -d '{"instances": [[6.8, 2.8, 4.8, 1.4]]}')
+
+        if [[ "$RESPONSE" == "403" ]]; then
+          echo "Test passed: Unauthenticated request correctly rejected with $RESPONSE"
+        else
+          echo "Test failed: Unauthenticated request returned $RESPONSE instead of 403"
+          exit 1
+        fi
 
-        echo "=== Detailed Curl Test ==="
-        curl -vv \
+    # Test 3: External access with token from unauthorized namespace, Must Fail
+    - name: Test external access with token from unauthorized namespace
+      run: |
+        export KSERVE_INGRESS_HOST_PORT=localhost:8080
+
+        # Create attacker namespace and service account
+        kubectl create ns kubeflow-user-example-com-attacker
+        kubectl create serviceaccount attacker-sa -n kubeflow-user-example-com-attacker
+        export ATTACKER_TOKEN="$(kubectl -n kubeflow-user-example-com-attacker create token attacker-sa)"
+
+        RESPONSE=$(curl -s -o /dev/null -w "%{http_code}" \
           -H "Host: isvc-sklearn.kubeflow-user-example-com.example.com" \
-          -H "Authorization: Bearer ${KSERVE_M2M_TOKEN}" \
+          -H "Authorization: Bearer ${ATTACKER_TOKEN}" \
+          -H "Content-Type: application/json" \
+          "http://${KSERVE_INGRESS_HOST_PORT}/v1/models/isvc-sklearn:predict" \
+          -d '{"instances": [[6.8, 2.8, 4.8, 1.4]]}')
+
+        if [[ "$RESPONSE" == "403" ]]; then
+          echo "Test passed: Unauthorized namespace token correctly rejected with $RESPONSE"
+        else
+          echo "Test failed: Unauthorized namespace token returned $RESPONSE instead of 403"
```

Comment:
Please no if else and many echoes, keep it simple. Just fail the whole pipeline if the test fails. That should work with curl --fail --show-errors